lib/ansible/modules/cloud/amazon/lightsail_lb.py

```diff
@@ -0,0 +1,272 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: lightsail_lb
+short_description: Creates or deletes load balancer and attaches running instances to the created load balancer in AWS Lightsail.
+description:
+     - Create or delete a load balancer in AWS Lightsail.
+version_added: "2.8"
+author: "Gencebay Demir (@gencebay)"
+options:
+  state:
+    description:
+      - Indicate desired state of the target.
+    default: present
+    choices: ['present', 'absent']
+  name:
+    description:
+      - The name of the load balancer.
+    required: true
+  region:
+    description:
+      - The AWS Region name.
+    required: true
+  attachInstances:
+    description:
+      - An array of strings representing the instance name(s) you want to attach to your load balancer.
+  port:
+    description:
+      - The instance port where you're creating your load balancer.
+    default: 80
+  healthCheckPath:
+    description:
+      - The path you provided to perform the load balancer health check.
+
+requirements:
+  - "python >= 2.6"
+  - boto3
+
+extends_documentation_fragment:
+  - aws
+  - ec2
+'''
+
+
+EXAMPLES = '''
+# Create a new load balancer
+- lightsail_lb:
+    state: present
+    name: loadbalancer1
+    region: eu-central-1
+    attachInstances: ["Ubuntu-512MB-1", "Ubuntu-512MB-2"]
+    port: 80
+    healthCheckPath: /api/health
+  register: lb_result
+
+- debug:
+    msg: "DNS name is {{ lb_result.loadbalancer.dns_name }}"
+  when: lb_result.loadbalancer.dns_name is defined
+
+- debug:
+    msg: "Protocol is {{ lb_result.loadbalancer.protocol }}"
+  when: lb_result.loadbalancer.protocol is defined
+
+# Delete the load balancer if present
+- lightsail_lb:
+    state: absent
+    region: eu-central-1
+    name: loadbalancer1
+
+'''
+
+RETURN = '''
+changed:
+  description: if a load balancer has been modified/created
+  returned: always
+  type: bool
+  sample:
+    changed: true
+loadbalancer:
+  description: load balancer info
+  returned: when state is present
+  type: dict
```

Comment:
This should be more specific about the contents of the data structure that's returned.

Reply:
Done